### PR TITLE
DDS create/delete_datawriter

### PR DESCRIPTION
**Pull request by laura-eprosima**
* Change DataWriter to use DataWriterQos instead of WriterQos
* create_datawriter standard implementation
* delete_datawriter standard implementation
* Add unit tests for DataWriter methods

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1948/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1685/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1942/)

File: src/cpp/fastdds/publisher/PublisherImpl.cpp [hunk 1]
```diff
@@ -257,18 +198,21 @@ DataWriter* PublisherImpl::create_datawriter(
...
+    fastrtps::TopicAttributes topic_att = DataWriterImpl::get_topic_attributes(qos, *topic, type_support);
     rtps_participant_->registerWriter(impl->writer_, topic_att, wqos);
 
     {
         std::lock_guard<std::mutex> lock(mtx_writers_);
-        writers_[topic_att.getTopicName().to_string()].push_back(impl);
+        writers_[topic->get_name()].push_back(impl);
     }
 
+    topic->get_impl()->reference();
```
Comment (IkerLuengo) [inline src/cpp/fastdds/publisher/PublisherImpl.cpp:214]: Move this before creating the datawriter, or some other thread may remove the Topic meanwhile.
Best solution would be some kind of locking on the Topic, though

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1958/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1691/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1952/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1965/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1698/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1959/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1969/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1702/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1963/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1974/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1707/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1968/)

**Comment by MiguelCompany** [comment]
Closing, as this has been included on #1156

**Pull request closed without merging**